compactor/periodic.go

```diff
@@ -107,6 +100,18 @@ func (t *Periodic) Run() {
 	}()
 }
 
+// If given compaction period x is <1-hour, compact every x duration, with x retention window
+// (e.g. --auto-compaction-mode 'periodic' --auto-compaction-retention='10m', then compact every 10-minute).
+// If given compaction period x is >1-hour, compact every 1/x duration, with x retention window
+// (e.g. --auto-compaction-mode 'periodic' --auto-compaction-retention='72h', then compact every 7.2-hour).
+func (t *Periodic) getInterval() time.Duration {
+	itv := t.period
+	if itv > time.Hour {
+		itv /= 10
```

Comment:
why is this changed? I thought we are going to do ` x is >1-hour` -> ` compact every hr`?

Reply:
This is to keep the old behavior. But at this point, we break old behavior anyways.

Reply:
```
if you set 60 minutes, compaction happens every 6 minutes
if you set 59 minutes, compaction happens every 59 minutes
```
The above behavior can be avoided if we do `x is >1-hour -> compact every hr after initial wait?`?

Reply:
As you pointed out https://github.com/coreos/etcd/pull/9474#discussion_r176218858, that would break old behavior. If we don't backport, I am fine with this too.

Reply:
The original code introduced an concept of checkInterval. This interval just to check if I should do compaction or not? it can be quick we had it set as 1/10 X . So If a compact error happens, we will just wait 1/10X to check again. We might want to keep that check interval. The interval in this pr is the compact interval which is different from the check interval.

Reply:
> This interval just to check if I should do compaction or not? it can be quick we had it set as 5 mins . So If a compact error happens, we will just wait another 5 mins to check again.

5-min interval is for revision compactor. So, v3.3.2 still retries after 1/10 of given compaction period. https://github.com/coreos/etcd/blob/v3.3.2/compactor/periodic.go#L64

Reply:
Discussed in person. @fanminshi meant 5-min in v3.2 :)